File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    
    h_list = []
    h_index = 0
    
    citations = sorted(citations)
    
    for i in range(len(citations)):
        cnt = 0
        for j in citations:
            if j >= citations[i]:
                cnt += 1
                
        h_index = min(cnt, citations[i])
        h_list.append(h_index)
            
    
    return max(h_list)

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int n = citations.length;
        Arrays.sort(citations);
        
        for (int i = 0; i < n; i++) {
            int h = n - i; // 이 위치에서 h 후보
            if (citations[i] >= h) return h; // 첫 만족하는 h가 최대 h-index
        }
        
        return 0;
    }
}